src/bounce_solution/bounce_solution.cpp

```diff
@@ -0,0 +1,1256 @@
+ * default set to 71 %
+ */
+void BounceSolution::CalculatePercolationTemp(const double &false_vac_frac)
+{
+  if (status_bounce_sol == "success")
+  {
+    Tperc = CalcTempAtFalseVacFraction(false_vac_frac);
+
+    if (Tperc > 0 and percolation_temp_set == false)
+    {
+      // Try to calculate action at Tp
+      // CalculateActionAt(Tperc);
+      for (std::size_t i = 0; i < SolutionList.size() - 1; i++)
+      {
+        if (Tperc > SolutionList[i].T and Tperc < SolutionList[i + 1].T)
+          CalculateActionAt((SolutionList[i].T + SolutionList[i + 1].T) / 2.);
+      }
+      percolation_temp_set = true;
+      SetBounceSol();
+      Tperc       = CalcTempAtFalseVacFraction(false_vac_frac);
+      status_perc = "success";
+    }
+    else if (Tperc < 0)
+    {
+      Logger::Write(LoggingLevel::TransitionDetailed,
+                    "Calculation of the percolation temperature failed.");
+      status_perc = "not_met";
+    }
+    return;
+  }
+  return;
+}
+
+/**
+ * @brief CalculateCompletionTemp calculation of the temperature when the false
+ * vacuum fraction drops below 1 % (default)
+ * @param false_vac_frac false vacuum fraction at completion temperature, by
+ * default set to 1 %
+ */
+void BounceSolution::CalculateCompletionTemp(const double &false_vac_frac)
+{
+  if (status_bounce_sol == "success")
+  {
+    Tcompl = CalcTempAtFalseVacFraction(false_vac_frac);
+
+    if (Tcompl > 0 and completion_temp_set == false)
+    {
+      for (std::size_t i = 0; i < SolutionList.size() - 1; i++)
+      {
+        if (Tcompl > SolutionList[i].T and Tcompl < SolutionList[i + 1].T)
+          CalculateActionAt((SolutionList[i].T + SolutionList[i + 1].T) / 2.);
+      }
+      completion_temp_set = true;
+      Tcompl              = CalcTempAtFalseVacFraction(false_vac_frac);
+      status_compl        = "success";
+    }
+    else if (Tcompl < 0)
+    {
+      Logger::Write(LoggingLevel::TransitionDetailed,
+                    "Calculation of the completion temperature failed.");
+      status_compl = "not_met";
+    }
+    return;
+  }
+  return;
+}
+
+double inner_integrand(double Temp, void *params)
+{
+  class BounceSolution &obj = *static_cast<BounceSolution *>(params);
+  double func               = 1. / obj.Hubble_Rate(Temp);
+  return func;
+}
+
+double outer_integrand(double Temp, void *params)
+{
+  class BounceSolution &obj = *static_cast<BounceSolution *>(params);
+
+  double func = obj.Tunneling_Rate(Temp) /
+                (std::pow(Temp, 4) * obj.Hubble_Rate(Temp)) *
+                std::pow(Nintegrate_Inner(obj, Temp).result, 3);
+  return func;
+}
+
+/**
+ * @brief Nintegrate_Inner Numerical integration of inner integral over
+ * inverse Hubble rate for the percolation temperature calculation
+ * @param obj Class reference to pass all needed parameters
+ * @param T upper integration boundary
+ * @return Numerical value of integral and absolute error
+ */
+struct resultErrorPair Nintegrate_Inner(BounceSolution &obj,
```

Comment:
Shouldnt the documentation go to the header?

Reply:
Done.